### DISCUSSION TITLE
Make undelegations available via SSE after 7 eras passed

### DISCUSSION THREAD

**Discussion by GuybrushX** [casper-network/casper-node#4076] [Updates]
Undelegation currently takes 7 eras. Using the SSE event stream it is possible to catch the undelegation deploy BUT according to my knowledge and research it's currently not possible to react on the event of receiving the amount in your liquid balance after that 7 eras. Currently this seems to be a manual process.

Example:
- I'm using the .Net or any other SDK and I'm listening to SSE events
- I undelegate 1000 CSPR right now -> I get a DeployAccepted and DeployProcessed event for the undelegation
- I wait 7 full eras -> the 1000 CSPR are eventually deposited on my wallet BUT I don't get any event for this

So according to my understanding I would need to keep track of undelegations myself and check the balance regularly and/or after 7 eras. At the same time I can't detect undelegations which happened already before I started to listen to events since I don't know they happened.

I would like to see this added to the protocol and to the SSE event stream to make it easier to react on such events. Are there any (technical) reasons why this isn't available already or am I just the first asking for this?

**Comment by KillianH** [comment]
I'm not tech savy enough on the node software to know how hard it would be to implement but on the event side since you already listen to them I guess you have some kind of data storage on your side. You could easily calculate the era where the funds would be available.
